app/controllers/submissions_controller.rb

```diff
@@ -357,6 +358,16 @@ def update_files
     if delete_files.empty? && new_files.empty? && new_folders.empty? && delete_folders.empty?
       flash_message(:warning, I18n.t('student.submission.no_action_detected'))
     else
+      if unzip
+        zdirs, zfiles = new_files.map do |f|
+          next unless f.content_type == 'application/zip'
```

Comment:
Hmm I guess this is a Windows thing, when I tried this out I got a content type of `'application/x-zip-compressed'`. I did a bit of research, and it looks like there are a few others. We could switch this to using the extension?